Xamarin.Forms.Core/Button.cs

```diff
@@ -23,6 +23,10 @@ public class Button : View, IFontElement, ITextElement, IBorderElement, IButtonC
 
 		public static readonly BindableProperty TextColorProperty = TextElement.TextColorProperty;
 
+		public static readonly BindableProperty HorizontalTextAlignmentProperty = BindableProperty.Create("HorizontalTextAlignment", typeof(TextAlignment), typeof(Button), TextAlignment.Center);
```

Comment:
should be `HorizontalTextAlignmentProperty = TextAlignmentElement.HorizontalTextAlignmentProperty` and the class should implement `ITextAlignmentElement`. This is required so the CSS engine can apply the `text-align` property.

Reply:
https://github.com/xamarin/Xamarin.Forms/blob/c18e85ee434d7bacb730215fecf15acc0230917f/Xamarin.Forms.Core/TextAlignmentElement.cs#L6

The type of owner class here is `EntryCell`. Won't that cause us some problems when using this property for other controls(i.e. in the Button control as you requested)?